### ISSUE TITLE
"Unexpected non-success response from Z3"

### ISSUE THREAD

**Issue by joelburget** [LeventErkok/sbv#342]
My program (testing [an example from the docs](https://hackage.haskell.org/package/sbv-7.5/docs/Data-SBV.html#v:solve)):

```haskell
main = do
  let sym = do [x, y, z] <- sIntegers ["x", "y", "z"]
               solve [x .> 5, y + z .< x]
  result <- runSMT sym
  print result
  print =<< isSatisfiable result
```

Result:

```
λ> main
<symbolic> :: SBool
*** Exception:
*** Data.SBV: Unexpected non-success response from Z3:
***
***    Sent      : (define-fun s1 () Int (+ s1 s2))
***    Expected  : success
***    Received  : (error "line 10 column 25: unknown constant s1")
***
***    Exit code : ExitFailure (-15)
***    Executable: /usr/local/bin/z3
***    Options   : -nw -in -smt2
***
***    Reason    : Check solver response for further information. If your code is correct,
***                please report this as an issue either with SBV or the solver itself!
```

Maybe helpful:

```
❯ z3 --version
Z3 version 4.5.0 - 64 bit
```

This is sbv-7.5.

Changing the `runSMT` line to `  result <- runSMTWith (defaultSMTCfg {verbose = True, redirectVerbose = Just "sbv-output", transcript = Just "sbv-transcript"}) sym` doesn't output to either file for some reason.

**Comment by LeventErkok** [comment]
Can you upgrade Z3 to v4.6.0 and give it a try?

**Comment by LeventErkok** [comment]
You should upgrade to z3 4.6.0.

But aside from that, the `runSMT` command is to be used when there's an associated call to `query` https://hackage.haskell.org/package/sbv-7.5/docs/Data-SBV-Control.html#v:query

I do agree that the error message is very misleading, however!

For your example, you should use a simple call to `sat`. Something like this:

```haskell
import Data.SBV

main = do
  let sym = do [x, y, z] <- sIntegers ["x", "y", "z"]
               solve [x .> 5, y + z .< x]
  result <- sat sym
  print result
  print =<< isSatisfiable sym
```

**Comment by joelburget** [comment]
Thanks for the help! Switching to `sat` fixed my issue, though fwiw upgrading z3 4.6 didn't change anything.

**Comment by LeventErkok** [comment]
Quite right.. The Z3 upgrade is still quite necessary as there were many other changes in 4.6.0 that you'd eventually need if you kept using 4.5.0.

Can we close this ticket? Feel free to open others as you run into issues.
